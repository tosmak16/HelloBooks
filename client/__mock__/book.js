const books = [
  {
    id: 1,
    bookId: 1,
    bookTitle: 'THE PEOPLE IN THE TREE',
    author: 'Y. Hayan',
    category: 'Adventure',
    isbn: '345798465',
    stocknumber: '2',
    image: 'l1.jpg',
    summary: 'Lorem Ipronic typesetting, remaininised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum',
  },
  {
    id: 2,
    bookId: 2,
    bookTitle: 'THE LATE PARADE',
    author: 'F. Adam',
    category: 'Adventure',
    isbn: '234560878',
    stocknumber: '0',
    image: 'l2.jpg',
    summary: 'It is a long established fact thaorem Ipsu uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like)',
  },
  {
    id: 3,
    bookId: 3,
    bookTitle: 'I AM THE WOLF',
    author: 'L. Mark',
    category: 'Music',
    isbn: '4678980976',
    stocknumber: '5',
    image: 'l3.jpg',
    summary: 'There are many variations of passages of Lorem Ipsum available, but the majority orem Ipsum generators on the Internet tend to repeat predefined chunks as necessary',

  },
  {
    id: 4,
    bookId: 4,
    bookTitle: 'THE PALE NORTH',
    author: 'H. Clayton',
    category: 'Adventure',
    isbn: '34567887655',
    stocknumber: '3',
    image: 'l4.jpg',
    summary: 'The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from ',
  },
  {
    id: 5,
    bookId: 5,
    bookTitle: 'ADULTOLESENCE',
    author: 'G. Hannah',
    category: 'Romance',
    isbn: '45678974567',
    stocknumber: '10',
    image: 'l5.jpg',
    summary: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Urum.'
  },
  {
    id: 6,
    bookId: 6,
    bookTitle: 'WHAT IT MEANS WHEN A MAN FALL FROM SKY',
    author: 'L. Nneka',
    category: 'Adventure',
    isbn: '567244567',
    stocknumber: '10',
    image: 'l6.jpg',
    summary: 'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium?'
  },
  {
    id: 7,
    bookId: 7,
    bookTitle: 'THE SONG AND THE SILENCE',
    author: 'Y. Johnson',
    category: 'Music',
    isbn: '64578974567',
    stocknumber: '10',
    image: 'l7.jpg',
    summary: 'But I must explain to you how all this mists no resultant pleasure  ducimus qui blanditiis praesentium voluptatum culpa qui officia ?'
  },
  {
    id: 8,
    bookId: 8,
    bookTitle: 'THE ILIAD HOMMER',
    author: 'P. Green',
    category: 'Action',
    isbn: '787890974567',
    stocknumber: '8',
    image: 'l8.jpg',
    summary: 'At vero eos et accusamus et iuse delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.'
  },
];

export default books;
