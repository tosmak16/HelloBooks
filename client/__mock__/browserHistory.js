import { createMemoryHistory } from 'react-router';


const browserHistory = createMemoryHistory('/');

export default browserHistory;
