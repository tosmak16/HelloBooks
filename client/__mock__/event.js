export const e = {
  target: {
    name: 'Tosmak',
    value: '10'
  },
};

export class event {
  static preventDefault() {

  }
}

