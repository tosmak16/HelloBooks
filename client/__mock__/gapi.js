const render = () => { };
const signin2 = {
  render,
};
global.gapi = {
  signin2
};
