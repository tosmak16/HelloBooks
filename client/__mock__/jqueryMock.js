import $ from 'jquery';
import React from 'react';

global.$ = global.jQuery = $;

$.prototype.modal = () => { };

global.Materialize = {
  toast: () => { }
};
