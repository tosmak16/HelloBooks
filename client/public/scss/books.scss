#helloImg{
  width: 20px;
  height: 20px;
}
.nav > li > a:hover, .nav > li > a:focus {
text-decoration: none;
  color: #fff;
  background: rgba(255,255,255,0.2);
}

#upper_nav{
  margin-top: 50px;
}
@media (min-width: 600px) {
#book_details_wrapper{
  margin-left: 300px;

}
}

#wishbtn{
background-color: #f0ad4e;
}

#borrowbtn{
background-color: #5cb85c;
}
#stockInput{
  width: 400px;
}
div #searchbarr{
  margin-top: 10px;
  margin-left: 80px;
}

#photo{
margin-right: 100px;
}

#sum{
width: 300px;
}

#create-btn{
  background-color: #3d1112 ;
  color: white;
  
}
.navbar{
  background-color: coral;
  
}
div #searchinpt{
  width: 150px;
}

#loginForm{
  margin-top: 100px;
  margin-bottom: 250px;
}

#sidenava{
width: 1000px;
}
@media (min-width: 600px) {
#sidenava {
  width: 200px;
  margin-top: 5px;
 
}
}

@media (max-width: 600px) {
  #sidenava {
    width: 600px;
   
  }
  }

#modal2,#modal1,#modal3, #modalOpen,#modalSuccess,#modalError, #modalO,#modalS,#modalE, #modaO,#modaS,#modaE {
    border: 20px;
    border-color: burlywood;
    opacity: 50%;
    color: #3d1112;
    border-style: double;
    background-color: wheat;
    border-width: 10px;
    border-radius: 50px;
    top: 25% !important;
    left: 28%!important;
    width: 450px;    
  }

  @media (max-width: 600px) {
    #modal2,#modal1,#modal3, #modalOpen,#modalSuccess,#modalError,#modalO,#modalS,#modalE, #modaO,#modaS,#modaE {
      border: 20px;
      border-color: burlywood;
      opacity: 50%;
      color: #3d1112;
      border-style: double;
      background-color: wheat;
      border-width: 10px;
      border-radius: 50px;
      top: 45% !important;
      right: 33%!important;
      left: auto!important;
      bottom: auto!important;
      width: 350px;  
      
    }
    }
  



  // #modal2{
  //   border: 20px;
  //   border-color: burlywood;
  //   opacity: 50%;
  //   color: #3d1112;
  //   border-style: double;
  //   background-color: wheat;
  //   border-width: 10px;
  //   border-radius: 50px;
  // }

  // #modal3{
  //   border: 20px;
  //   border-color: burlywood;
  //   opacity: 50%;
  //   color: #3d1112;
  //   border-style: double;
  //   background-color: wheat;
  //   border-width: 10px;
  //   border-radius: 50px;
  // }


@media (min-width: 600px) {
#book_body{
  padding-left: 50px;
  margin-left: 50px;

  
 

}
}

.collapsible{
  color: #3d1112;
}

#card_div{
padding-right: 10px;

}

#card_box{
  padding-right: 10px;
  width: 150px;
}
#card_box_details{
  padding-right: 10px;
  width: 150px;
}
.card_holder{
  width: 150px;
}
.card_holder_details{
  width: 150px;
  height: 500px;
}




.jumbotron p{
  color:white;
}
h2{
  color: whitesmoke;
}
#readSummary {
  text-align: center;
}
#hello{
  font-size: x-large;
  color: wheat;
  
}
#navitem{
  font-size: medium;
}

#searchbarr{
  margin-top: 1px;
  
}
#searchinput{
  width: 130px;
  box-sizing: border-box;
  border: 1px solid #ccc;
  border-radius: 4px;
  font-size: 12px;
  background-color: white;
  background-position: 10px 10px; 
  background-repeat: no-repeat;
  padding: 12px 20px 12px 40px;
  -webkit-transition: width 0.4s ease-in-out;
  transition: width 0.4s ease-in-out;
}
#searchbtn{
  height: 32px;
}
#booksearchbtn{
  background-color: gainsboro;

}

/* ========================== book page style==============  */

/*
* Base structure
*/

/* Move down content because we have a fixed navbar that is 50px tall */



a.heada{
color: burlywood;

}

a.boda{
color: wheat;


}


div.vol{
float: left;
}

#booksearchinput{
    width: 270px;

}
  #adminsearchinput{
    width: 150px;

}
  

/*
* Global add-ons
*/

.sub-header {
padding-bottom: 10px;
border-bottom: 1px solid #eee;
}

/*
* Top navigation
* Hide default border to remove 1px line.
*/
.navbar-fixed-top {
border: 0;
}

/*
* Sidebar
*/

/* Hide for mobile, show later */
.sidebar {
display: block;

top: 51px;
bottom: 0;
left: 0;

padding: 20px;
padding-top: 100px; 
background-color: #3d1112;
}
@media (min-width: 600px) {
.sidebar {
  position: fixed;
  top: 51px;
  bottom: 0;
  left: 0;
  z-index: 1000;
  display: block;
  padding: 20px;
  
  background-color: #3d1112;
 
}
}

/* Sidebar navigation */
.nav-sidebar {
/* 20px padding + 1px border */
margin-bottom: 20px;
margin-left: -20px;

width: 500px;

}
.nav-sidebar > li > a {
padding-right: 0px;
padding-left: 20px;
}
.nav-sidebar > .active > a,
.nav-sidebar > .active > a:hover,
.nav-sidebar > .active > a:focus {
color: #fff;
background-color: #3d1112;

}


/*
* Main content
*/

.main {
padding: 20px;
}
@media (min-width: 768px) {
.main {
  padding-right: 40px;
  padding-left: 40px;
}
}
.main .page-header {
margin-top: 5;
}


/*
* Placeholder dashboard ideas
*/

.placeholders {
margin-bottom: 30px;
text-align: center;
}
.placeholders h4 {
margin-bottom: 0;
}
.placeholder {
margin-bottom: 20px;
}
.placeholder img {
display: inline-block;

}

/*======book details style=================*/

.dplaceholder{
  float: left;
  margin-right: 150px;
  margin-left: 100px;
  margin-top: 10px;
  padding-right: 20px;
}
.summary{
 
 margin-left: 10px;
  margin-top: 10px;
}
.shop{
  width: 150px;
  
}

.div-p{
  margin-top: 20px;
}

.bookinfo{
  color: grey;
}
 
 .heada{
  color: wheat;
}


