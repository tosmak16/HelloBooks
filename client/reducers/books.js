import { GET_ALL_BOOKS_FAILURE, GET_ALL_BOOKS_SUCCESS, GET_ALL_BOOKS_REQUEST } from '../actions/getBooks';

const initialState = {
  isFetching: false,
  isFetched: false,
  data: [],
  error: '',
};
export default (state = initialState, action = {}) => {
  switch (action.type) {
    case GET_ALL_BOOKS_REQUEST:
      return {
        isFetching: true,
        isFetched: false,
        data: [],
        error: '',
      };
    case GET_ALL_BOOKS_SUCCESS:
      return {
        isFetching: false,
        isFetched: true,
        data: action.data,
        error: '',
      };
    case GET_ALL_BOOKS_FAILURE:
      return {
        isFetching: false,
        isFetched: false,
        data: [],
        error: action.error,
      };
    default:
      return state;
  }
};
