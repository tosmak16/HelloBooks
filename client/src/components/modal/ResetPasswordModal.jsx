import React from 'react';
import PropTypes from 'prop-types';

/**
 * @function ResetPasswordModal
 * @param {object} props
 * @returns {view} modal
 */
export const ResetPasswordModal = (props) => {
  const { handleInputEmailChange, handleClose, handleSendMail, userEmailText } = props;
  return (
    <div id="modalPassword" className="modal">
      <button
        id="modal-cancel"
        onClick={handleClose}
        className="material-icons brown-text"
      >
        cancel
      </button>
      <div id="modalPasswordHeader">
        <h5 style={{ color: 'indianred' }} >Forgotten your password?</h5>
      </div>
      <div className="modal-content">
        <p>We will send you a password reset mail  to use for login.
         Enter your login email address below
        </p>
        <div className="input-field ">
          <label htmlFor="firstname" className="sr-only">First Name</label>
          <i style={{ top: '20px' }} className="material-icons brown-text prefix">contact_mail</i>
          <input
            id="firstname"
            onChange={handleInputEmailChange}
            className="form-control validate"
            required
            value={userEmailText}
            type="email"
            name="email"
          />
          <div>
            <button
              onClick={handleSendMail}
              id="modalButton"
              className="waves-effect waves-brown material-icons"
            >
              send
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};
ResetPasswordModal.propTypes = {
  handleClose: PropTypes.func.isRequired,
  handleInputEmailChange: PropTypes.func.isRequired,
  handleSendMail: PropTypes.func.isRequired,
  userEmailText: PropTypes.string.isRequired
};
export default ResetPasswordModal;
