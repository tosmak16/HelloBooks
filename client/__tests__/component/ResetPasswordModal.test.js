import expect from 'expect';
import { mount, shallow } from 'enzyme';
import render from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import React from 'react';
import $ from 'jquery';
import localStorageMock from '../../__mock__/localStorage';
import books from '../../__mock__/book';
import { e, event } from '../../__mock__/event';
import { ResetPasswordModal } from '../../src/components/modal/ResetPasswordModal';

const middlewares = [thunk]; // add your middlewares like `redux-thunk
const mockFuction = jest.fn();
const mockStore = configureMockStore(middlewares);
let store;
let wrapper;
describe('Test book Details page components and container', () => {
  it('should render and take snapshot of ResetPasswordModal', () => {
    const tree = render.create(<ResetPasswordModal
      userEmailText=''
      handleSendMail={mockFuction}
      handleClose={mockFuction}
      handleInputEmailChange={mockFuction}
    />);
    expect(tree).toMatchSnapshot();
  });
});
