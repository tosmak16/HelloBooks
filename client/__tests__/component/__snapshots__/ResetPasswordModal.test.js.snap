// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test book Details page components and container should test and take snapshot of ResetPasswordModal 1`] = `
<div
  className="modal"
  id="modalPassword"
>
  <button
    className="material-icons brown-text"
    id="modal-cancel"
    onClick={[Function]}
  >
    cancel
  </button>
  <div
    id="modalPasswordHeader"
  >
    <h5
      style={
        Object {
          "color": "indianred",
        }
      }
    >
      Forgotten your password?
    </h5>
  </div>
  <div
    className="modal-content"
  >
    <p>
      We will send you a password reset mail  to use for login. Enter your login email address below
    </p>
    <div
      className="input-field "
    >
      <label
        className="sr-only"
        htmlFor="firstname"
      >
        First Name
      </label>
      <i
        className="material-icons brown-text prefix"
        style={
          Object {
            "top": "20px",
          }
        }
      >
        contact_mail
      </i>
      <input
        className="form-control validate"
        id="firstname"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
      <div>
        <button
          className="waves-effect waves-brown material-icons"
          id="modalButton"
          onClick={[Function]}
        >
          send
        </button>
      </div>
    </div>
  </div>
</div>
`;
